pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -118,6 +121,23 @@ public List<String> listSchemaNames(@Context HttpHeaders headers) {
     return _pinotHelixResourceManager.getSchemaNames(headers.getHeaderString(DATABASE));
   }
 
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/schemas/info")
+  @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_SCHEMA_INFO)
+  @ApiOperation(value = "List all schemas info with count of field specs", notes = "Lists all schemas with field "
+      + "count details")
+  public List<SchemaInfo> getSchemaInfo(@Context HttpHeaders headers) {
+    List<SchemaInfo> schemasInfo = new ArrayList<>();
+    TableCache tableCache = _pinotHelixResourceManager.getTableCache();
+    List<Schema> schemaList = tableCache.getSchemas();
+    for (Schema schema : schemaList) {
+      schemasInfo.add(new SchemaInfo(schema.getSchemaName(), schema.getDimensionFieldSpecs().size() - 3,
```

Comment:
Suggest adding a constructor to `SchemaInfo` which takes a `schema` and hide the implementation details. Also consider adding some comments describing why we deduct 3 from dimensions

Reply:
So, I noticed that some schemas don't have these virtual columns added. So adding logic to check specifically for each of the virtual column in a schema.

Reply:
Interesting.. Did you find when the virtual columns are missing? In `TableCache.putSchema()`, virtual columns are always first added before adding to the cache

Reply:
Sorry, I mis checked it with _pinotHelixResourceManager instead of tableCache. With tableCahe, yes it adds virtual columns in putSchema for each schema. Added comment and hid implementation as suggested.